Hazelcast.Net/Hazelcast.Core/VectorClock.cs

```diff
@@ -0,0 +1,68 @@
+﻿/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+using TimeStampIList = System.Collections.Generic.IList<System.Collections.Generic.KeyValuePair<string, long>>;
+
+namespace Hazelcast.Core
+{
+    internal class VectorClock
+    {
+        private readonly Dictionary<string, long> _timeStampDictionary = new Dictionary<string, long>();
+        private readonly TimeStampIList _timeStampList = new List<KeyValuePair<string, long>>();
+
+        public VectorClock()
+        {
+        }
+
+        public VectorClock(TimeStampIList timeStampList)
+        {
+            _timeStampList = timeStampList;
```

Comment:
I did it for performance optimization. Instead of (re)creating original list of timestamps every time from map/dictionary I send back original timestamps used for map/dictionary construction. It should be more efficient than creating new list on every call to EntrySet() method

Reply:
This is a pointer assignment, right? Does this pointer assignment avoid responseparameter object being not destroyed?